tests/integration/test_matrix_power.py

```diff
@@ -22,18 +22,88 @@
 import cunumeric as num
 
 # TODO: add negative exponents here, once they become supported
-EXPONENTS = [0, 1, 3, 5]
+EXPONENTS = (0, 1, 2, 3, 5)
 
 
+@pytest.mark.parametrize(
+    "dtype",
+    (
+        np.float64,
+        np.complex128,
+        pytest.param(np.int32, marks=pytest.mark.xfail),
+    ),
+)
 @pytest.mark.parametrize("ndim", range(0, LEGATE_MAX_DIM - 2))
 @pytest.mark.parametrize("exp", EXPONENTS)
-def test_matrix_power(ndim, exp):
+def test_matrix_power(ndim, exp, dtype):
+    # If dtype=np.int32 and exp greater than 1,
+    # In Numpy, pass
+    # In cuNumeric, raises TypeError: Unsupported type: int32
     shape = (3,) * ndim + (2, 2)
+    a_np = mk_0to1_array(np, shape, dtype=dtype)
+    a_num = mk_0to1_array(num, shape, dtype=dtype)
+    res_np = np.linalg.matrix_power(a_np, exp)
+    res_num = num.linalg.matrix_power(a_num, exp)
+    assert allclose(res_np, res_num)
+
+
+@pytest.mark.parametrize(
+    "exp",
+    (
+        0,
+        1,
+        pytest.param(2, marks=pytest.mark.xfail),
+        pytest.param(3, marks=pytest.mark.xfail),
+    ),
+)
+def test_matrix_power_empty_matrix(exp):
+    # If exp =2 or 3,
+    # In Numpy, pass and returns empty array
+    # In cuNumeric, raise AssertionError in _contract
+    shape = (0, 0)
     a_np = mk_0to1_array(np, shape)
     a_num = mk_0to1_array(num, shape)
     res_np = np.linalg.matrix_power(a_np, exp)
     res_num = num.linalg.matrix_power(a_num, exp)
-    assert allclose(res_np, res_num)
+    assert np.array_equal(res_np, res_num)
+
+
+class TestMatrixPowerErrors:
+    @pytest.mark.parametrize("ndim", (0, 1), ids=lambda ndim: f"(ndim={ndim})")
+    def test_matrix_ndim_smaller_than_two(self, ndim):
+        # In Numpy, it raises numpy.linalg.LinAlgError
```

Comment:
If numpy is raising `LinAlgError` that's definitely on purpose (i.e. we *do* definitely want to match exception *type* then) so `xfail` here (and anywhere else that misses raising `LinAlgError`)

Reply:
I fixed the source code so that the tests pass now.